.github/workflows/build_and_test.yml

```diff
@@ -0,0 +1,251 @@
+name: Build and Test
+
+on:
+  push:
+    branches:
+      - '**'
+  pull_request:
+    types: [opened, reopened]
+    branches:
+      - '**'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    container:
+      image: openjdk:8-jdk
+    defaults:
+      run:
+        working-directory: .
+    steps:
+      - uses: actions/checkout@v4
+      - name: Setup System Tools
+        run: |
+          apt update -y
+          apt install -y gnupg2 curl
+
+      - name: Verify files
+        run: |
+          curl -sSL https://secchannel.rsk.co/SUPPORT.asc | gpg2 --import -
+          gpg2 --verify SHA256SUMS.asc && sha256sum --check SHA256SUMS.asc
+
+      - uses: actions/cache@v4
+        name: Cache Gradle
+        id: cache-gradle
+        with:
+          path: |
+            .gradle/caches
+            gradle/wrapper
+            rskj-core/build
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties', 'rskj-core/src/main/resources/version.properties') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+
+      - name: Get gradle wrapper and build dependencies
+        if: steps.cache-gradle.outputs.cache-hit != 'true'
+        run: |
+          ./configure.sh
+          ./gradlew --no-daemon dependencies
+
+      - name: Build
+        run: |
+          ./gradlew --no-daemon --stacktrace build -x test
+
+      - name: Archive build artifacts
+        uses: actions/upload-artifact@v4
+        with:
+          name: build-files
+          path: |
+            rskj-core/build
+
+      - name: List the built jars
+        run: |
+          find rskj-core/build/libs/ -name "*.jar"
+
+  sonarqube:
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Setup Java JDK
+        uses: actions/setup-java@v3
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+
+      - uses: actions/cache@v4
+        name: Restore Gradle cache
+        with:
+          path: |
+            .gradle/caches
+            gradle/wrapper
+            rskj-core/build
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties', 'rskj-core/src/main/resources/version.properties') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+
+      - name: Download build artifacts
+        uses: actions/download-artifact@v4
+        with:
+          name: build-files
+          path: |
+            rskj-core/build
+
+      - name: List the built artefacts
+        run: |
+          tree rskj-core/build -L 2  
+
+      - name: Run SonarQube analysis
+        run: |
+          extra_flags=""
+          if [[ "${{ github.event_name }}" == "pull_request" ]]; then
```

Comment:
why not using a github action condition instead of an if on bash?

The block should look like:

```
- name: If pull request
  if: github.event_name == 'pull_request'
  run: |
    git branch -f master origin/master
    pr_number==${{ github.event.pull_request.number }}
    extra_flags="-Dsonar.pullrequest.base=master
      -Dsonar.pullrequest.branch=${{ github.head_ref }}
       -Dsonar.pullrequest.key=$pr_number"

- name: if not pull request
  if: github.event_name != 'pull_request'
  run: extra_flags="-Dsonar.branch.name=master"

- name: Run SonarQube analysis
  run: |
              ./gradlew sonarqube --no-daemon -x build -x test \
              $extra_flags \
              -Dsonar.organization=rsksmart \
              -Dsonar.host.url="https://sonarcloud.io" \
              -Dsonar.token="${{ secrets.SONAR_TOKEN }}"
              
```

Reply:
done